app/app.go

```diff
@@ -77,26 +78,52 @@ func NewChildChain(logger log.Logger, db dbm.DB) *ChildChain {
 	return app
 }
 
+func (app *ChildChain) initChainer(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
+	stateJSON := req.AppStateBytes
+	// TODO is this now the whole genesis file?
+
+	var genesisState GenesisState
+	err := app.cdc.UnmarshalJSON(stateJSON, &genesisState)
+	if err != nil {
+		panic(err) // TODO https://github.com/cosmos/cosmos-sdk/issues/468
+		// return sdk.ErrGenesisParse("").TraceCause(err, "")
+	}
+
+	// load the accounts
+	for _, gutxo := range genesisState.UTXOs {
+		utxo := ToUTXO(gutxo)
+		app.utxoMapper.AddUTXO(ctx, utxo)
+	}
+
+	// load the initial stake information
+	return abci.ResponseInitChain{}
+}
+
+// RLP decodes the txBytes to a BaseTx
 func (app *ChildChain) txDecoder(txBytes []byte) (sdk.Tx, sdk.Error) {
-	// TODO: implement method with RLP
 	var tx = types.BaseTx{}
-	// BaseTx is struct for Msg wrapped with authentication data
-	err := app.cdc.UnmarshalBinary(txBytes, &tx)
+
+	err := rlp.DecodeBytes(txBytes, &tx)
 	if err != nil {
-		return nil, sdk.ErrTxDecode("")
+		return nil, sdk.ErrTxDecode(err.Error())
 	}
 	return tx, nil
 }
 
-func (app *ChildChain) endBlocker(ctx sdk.Context, req abci.RequestEndBlock) {
```

Comment:
Why was this removed, and what happened to ChildChain.txHash?

Reply:
My understanding is that only data within a KVStore is contained within the state hash. So txHash, feeAmount, and txIndex would not be included in such a state and therefore we don't need the endBlocker just yet. We will need to add it back when we decide on how to store the feeAmounts for each block. One simple way would be to just create a store mapping from block number to feeAmount, but on the rootchain then you would need to prove the existence of the feeAmount within the store and then the existence of the store within the state hash.

Is this correct @AdityaSripal ?

Reply:
@colin-axner  correct.

The only data that gets written to appHash is the data in the commit multistore (app.cms). The only way to put data into the commit multistore is to set the data into some store that is part of the multistore.

The reason txHash was taken out was because it never got set in a store, so it was never part of the appHash.

Reasons why it wasn't added back correctly yet:
Correct inclusion of txHash is a problem for 0.3.0
Still thinking about the best way to include txHash into the appHash.